python/semantic_kernel/data/vector_search/vector_search.py

```diff
@@ -0,0 +1,107 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import logging
+from abc import abstractmethod
+from collections.abc import AsyncIterable
+from typing import Any, Generic, TypeVar
+
+from semantic_kernel.data.kernel_search_results import KernelSearchResults
+from semantic_kernel.data.search_options import SearchOptions
+from semantic_kernel.data.vector_search.vector_search_options import VectorSearchOptions
+from semantic_kernel.data.vector_search.vector_search_result import VectorSearchResult
+from semantic_kernel.data.vector_storage.vector_store_record_collection import VectorStoreRecordCollection
+from semantic_kernel.utils.experimental_decorator import experimental_class
+
+TModel = TypeVar("TModel")
+TKey = TypeVar("TKey")
+
+logger = logging.getLogger(__name__)
+
+
+@experimental_class
+class VectorSearchBase(VectorStoreRecordCollection[TKey, TModel], Generic[TKey, TModel]):
+    """Method for searching vectors."""
+
+    @property
+    def options_class(self) -> type[SearchOptions]:
+        """The options class for the search."""
+        return VectorSearchOptions
+
+    # region: New abstract methods to be implemented by vector stores
+
+    @abstractmethod
+    async def _inner_search(
+        self,
+        options: VectorSearchOptions | None = None,
+        search_text: str | None = None,
+        vectorizable_text: str | None = None,
+        vector: list[float | int] | None = None,
+        **kwargs: Any,
+    ) -> KernelSearchResults[VectorSearchResult[TModel]]:
+        """Inner search method.
+
+        This is the main search method that should be implemented, and will be called by the public search methods.
+        Currently, one of the three search contents will be provided, in the future,
+        this may be expanded to allow multiple of them.
```

Comment:
If we allowed multiple of them to be provided, do they do that by overriding the methods from the mixins?

Reply:
yes, their will then be new mixins for the new search types and those might call new params, or a combination of existing ones or existing and new.